pkg/machinepool/machinepool_test.go

```diff
@@ -225,6 +227,49 @@ var _ = Describe("Machinepool and nodepool", func() {
 				),
 			)
 		})
+		It("should return an error if both `subnet` and `availability-zone` flags are set", func() {
```

Comment:
This context is getting too large, lets create another one for just testing creating node pools

```suggestion
		Context("Create NodePool", func() {
			When("both `subnet` and `availability-zone` flags are set", func() {
				It("should return an error", func() {
					....
				})
			})
		})
```
Lets also make use of `When` and `It`, to help with readability

Reply:
I'll remove these tests for now and add them as part of a seperate pr with the new service changes